pkg/config/validation/extensionprovider.go

```diff
@@ -0,0 +1,120 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package validation
+
+import (
+	"fmt"
+	"net/url"
+	"strconv"
+	"strings"
+
+	envoytypev3 "github.com/envoyproxy/go-control-plane/envoy/type/v3"
+	"github.com/hashicorp/go-multierror"
+
+	meshconfig "istio.io/api/mesh/v1alpha1"
+)
+
+func validateExtensionProviderService(service string) error {
+	if service == "" {
+		return fmt.Errorf("service must not be empty")
+	}
+	parts := strings.Split(service, "/")
+	if len(parts) > 2 {
+		return fmt.Errorf("service value %q not in format [<namespace>/]<service>", service)
+	}
+	return nil
+}
+
+func validateExtensionProviderEnvoyExtAuthzStatusOnError(status string) error {
+	if status == "" {
+		return nil
+	}
+	code, err := strconv.ParseInt(status, 10, 32)
+	if err != nil {
+		return fmt.Errorf("invalid statusOnError value %s: %v", status, err)
```

Comment:
The func is passing a `status` string, without knowing it's actually passing the `x.status_on_error` field. the function naming seems is leaking details how how it should be used.

Reply:
I think it's fine, this is a local function (not exported) and is only supposed to be used in this package.